packages/paste-website/src/components/site-wrapper/site-header/DarkModeToggle.tsx

```diff
@@ -1,111 +1,62 @@
-import * as React from 'react';
-import {styled, css} from '@twilio-paste/styling-library';
-import {useUID} from '@twilio-paste/uid-library';
 import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
-import {Box} from '@twilio-paste/box';
-import {Stack} from '@twilio-paste/stack';
+import {Menu, MenuButton, MenuItemRadio, useMenuState} from '@twilio-paste/menu';
 import {DarkModeIcon} from '@twilio-paste/icons/esm/DarkModeIcon';
 import {LightModeIcon} from '@twilio-paste/icons/esm/LightModeIcon';
 
 import {useDarkModeContext} from '../../../context/DarkModeContext';
 import {event} from '../../../lib/gtag';
 
-const StyledThemeSwitcherLabel = styled.label<{checked: boolean}>((props) =>
-  css({
-    cursor: 'pointer',
-    display: 'inline-block',
-    flexGrow: 1,
-    justifyContent: 'center',
-    fontSize: 'fontSize30',
-    fontWeight: props.checked ? 'fontWeightBold' : 'fontWeightNormal',
-    lineHeight: 'lineHeight30',
-    paddingTop: 'space20',
-    paddingBottom: 'space20',
-    paddingRight: 'space40',
-    paddingLeft: 'space40',
-    backgroundColor: props.checked ? 'colorBackgroundStrongest' : 'transparent',
-    borderRadius: 'borderRadius10',
-    color: props.checked ? 'colorTextWeakest' : 'colorTextInverse',
-    '&:hover': {
-      textDecoration: 'underline',
-    },
-    '&:focus-within': {
-      boxShadow: 'shadowFocusInverse',
-      textDecoration: 'underline',
-    },
-  })
-);
-
-const StyledThemeSwitcherRadio = styled.input({
-  opacity: 0,
-  position: 'absolute',
-});
-
 export const DarkModeToggle = (): JSX.Element => {
   const {theme, toggleMode} = useDarkModeContext();
-  const lightModeID = useUID();
-  const darkModeID = useUID();
+  const menu = useMenuState();
 
-  const handleChange: React.ChangeEventHandler<HTMLInputElement> = () => {
+  const handleClick = (category: string, action: string, label: string): void => {
+    menu.hide();
     toggleMode();
+    event({
+      category,
+      action,
+      label,
+    });
   };
 
   return (
-    <Box display="flex">
-      <Box
-        as="form"
-        backgroundColor="colorBackgroundInverse"
-        borderColor="colorBorderInverse"
-        borderRadius="borderRadius20"
-        borderStyle="solid"
-        borderWidth="borderWidth10"
-        padding="space10"
-        maxWidth="size30"
+    <>
+      <MenuButton
+        {...menu}
+        variant="secondary_icon"
+        size="reset"
+        onClick={() =>
+          event({
+            category: 'Top Navigation',
+            action: 'click-dark-mode',
+            label: 'Switch',
```

Comment:
thought: I don't think tracking the menu open is meaningful here. If we really want it, label should be `MenuOpen` instead I think

Reply:
I was working on the assumption that there are three metrics. Did you interact with the toggle, did you choose light mode, did you choose dark mode. Keep the label means we can track any movement in the metric either way without too much data wrangling, the trend will just change. But equally, I don't think we really check the metrics